### PR TITLE
Remove `paper/delivery` url

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What are you doing in this PR?
- Removes the use of using a fully qualified route `paper/delivery` to control a fragment of the page in favour of using the url fragment/hash to do so
- Moves redirection for the paper route to Fastly

## Why are you doing this?

Using a URL to control a fragment of the page felt like the wrong tool for the job. This code also created some interesting logic between the history manipulation on the client and routing on the server where you couldn't hit `/paper` directly without redirecting to `/paper/delivery`.

It hopefully simplifies this part of the code too by using JS to deal with the fragment/hash and the server / fastly to deal with routing.

This code was introduced [here](https://github.com/guardian/support-frontend/pull/5436/files#diff-262aa4c1d3a23544743205a3065e7217fcc8f4638bf30c751a12469bd00e2f58R59). It made sense at the time, but hopefully this helps reduce out route fragmentation.

### SEO

I have gone with the `/paper` over `/paper/delivery` as `/paper/delivery` doesn't even feature in our Search Console. [Nor does it show up on Google SERPs](https://www.google.com/search?q=https%3A%2F%2Fsupport.theguardian.com%2Fuk%2Fsubscribe%2Fpaper%2Fdelivery).

# Screen-recording

https://github.com/guardian/support-frontend/assets/31692/f39f0c73-605a-4328-a8e9-91df509d8fab

### CODE REVIEW

File: support-frontend/conf/fastly-snippets/redirects-table.vcl [hunk 1]
```diff
@@ -49,4 +49,10 @@ table support_frontend_redirects {
   "/int/subscribe": "/int/subscribe/weekly",
   "/nz/subscribe": "/nz/subscribe/weekly",
   "/ca/subscribe": "/ca/subscribe/weekly",
+
+  # paper
+  "/paper": "/uk/subscribe/paper",
+  "/subscribe/paper": "/uk/subscribe/paper",
+  # This was a legacy URL to differentiate between tabs on the page
+  "/uk/subscribe/paper/delivery": "/uk/subscribe/paper",
```
Comment (jamesgorrie) [inline support-frontend/conf/fastly-snippets/redirects-table.vcl:57]: I'm half tempted to remove this as it doesn't feature anywhere and by supporting the redirect we offer it as an option.

File: support-frontend/assets/helpers/urls/routes.ts [hunk 2]
```diff
@@ -71,8 +70,10 @@ function paperSubsUrl(
 		.map((keyValuePair) => keyValuePair.join('='))
 		.join('&');
 
+	const hash = withDelivery ? `#${HomeDelivery}` : '';
```
Comment (andrewHEguardian) [inline support-frontend/assets/helpers/urls/routes.ts:73]: is this still necessary if the page always shows home delivery first anyway?

Comment (jamesgorrie) [reply support-frontend/assets/helpers/urls/routes.ts:73]: Good shout, I have decided to make the function more explicit that you can either send it a `PaperFulfilmentOptions` which links to the tabs, or `undefined`, which just links to the page.

Comment (andrewHEguardian) [reply support-frontend/assets/helpers/urls/routes.ts:73]: in a future PR? (or did I miss something sorry)

**Review by andrewHEguardian — APPROVED** [review]
Looks good, a nice way to handle the tabs 👍

File: support-frontend/assets/helpers/urls/routes.ts (hunk 2 above)
Comment (jamesgorrie) [reply support-frontend/assets/helpers/urls/routes.ts]: Nope, [I missed something](https://github.com/guardian/support-frontend/pull/5572/commits/0fc371a6b09e7dfde4a6cfbef85f563f9165ae54) (forgot to push).

**Pull request merged**